privacyidea/api/validate.py

```diff
@@ -201,6 +201,7 @@ def offlinerefill():
 @postpolicy(check_tokeninfo, request=request)
 @postpolicy(check_tokentype, request=request)
 @postpolicy(check_serial, request=request)
+@postpolicy(token_serial_from_response_g, request=request)
```

Comment:
Do we also need this at the endpoint ``triggerchallenge``?

Reply:
I first thought no, but I think we need it.

My thought: In the next step, you may have an admin policy "triggerchallenge" with a tokeninfo for the triggeradmin.
If the admin now triggers a challenge with serial, before_after will set g.serial -> no decorator needed.
If you trigger the challenge without serial for a specific user, you would like to have the policy condition tokeninfo act on this action. -> decorator needed.

Reply:
Triggerchallenge is now decorated twice as pre and post, since the serial may be in the request or not.

Actually, if the triggerchallenge has a serial in the request, we may end up with NO serial in the response. To not override a given serial by None, the g.serial is only set by the post-decoration if it was NOT set before. I cannot think of any case, where this is a problem. Can you?